src/Compiler/Checking/NameResolution.fs

```diff
@@ -2625,8 +2628,45 @@ let rec ResolveLongIdentInTypePrim (ncenv: NameResolver) nenv lookupKind (resInf
                 // make sure to keep the intrinsic pinfos before the extension pinfos in the list,
                 // since later on this logic is used when giving preference to intrinsic definitions
                 match DecodeFSharpEvent (pinfos@extensionPropInfos) ad g ncenv m with
-                | Some x -> success [resInfo, x, rest]
-                | None -> raze (UndefinedName (depth, FSComp.SR.undefinedNameFieldConstructorOrMember, id, NoSuggestions))
+                | Some x ->
+                    if
+                        not (g.langVersion.SupportsFeature LanguageFeature.PreferExtensionMethodOverPlainProperty)
+                        || not (List.isEmpty rest) // if not empty, it is .X.Y...
+                    then
+                        success [resInfo, x, rest]
+                    else
+                        match maybeAppliedArgExpr with
+                        | None -> success [resInfo, x, rest]
+                        | Some _argExpr ->
+                            // RFC-1137 prefer extension method when ...
+        
+                            let ignoreProperty (p: PropInfo) =
+                                // do not hide properties if:
+                                // * is indexed property
+                                // * is function type
+                                if p.IsIndexer then
+                                    true
+                                else
+                                    match p.GetPropertyType(ncenv.amap, m) with
+                                    | TType_var(typar={typar_solution = Some (TType_fun _) }) ->
+                                        true
+                                    | _ -> false
+                            
+                            match x with
+                            | Item.Property(info=ps) when ps |> List.exists ignoreProperty ->
+                                success [resInfo, x, rest]
+                            | _ ->
+                                // lookup in-scope extension methods
+                                // to keep in sync with the same expression in `| Some(MethodItem msets) when isLookupExpr` below
+                                match ExtensionMethInfosOfTypeInScope ResultCollectionSettings.AllResults ncenv.InfoReader nenv optFilter isInstanceFilter m ty with
+                                | [] -> success [resInfo, x, rest]
+                                | methods ->
+                                    let extensionMethods = Item.MakeMethGroup(nm, methods)
+                                    success ((resInfo,extensionMethods,rest)::[resInfo,x,rest])
+                | None ->
+                    // todo: consider if we should check extension method, but we'd probably won't have matched
```

Comment:
Can you elaborate on this? What could be a good code example for this?

Reply:
I've added a comment on the code which I didn't change, more as self note to check if code goes there with extension methods, maybe we could have an assert; the behaviour hasn't and shouldn't change due to this feature AFAIU. 

I don't know for sure a code sample that goes there unfortunately, it is related to having found properties matching the name in `TryFindIntrinsicNamedItemOfType` and `ExtensionPropInfosOfTypeInScope`, and the concat of those two returning an empty list, which result in `DecodeFSharpEvent` returning `None`, it would mean upstream is sending garbage basically.

Let me know if an assert sounds good and if it should have an assert at this point.

Reply:
Maybe we could leave things as they are, if not affected by this change.

Reply:
ok, letting it be like this, if you feel the comment doesn't add value, or is incorrect, feel free to adjust it.